hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/client/HddsClientUtils.java

```diff
@@ -132,9 +132,17 @@ private static boolean isSupportedCharacter(char c, boolean isStrictS3) {
     // ozone allows namespace to follow other volume/bucket naming convention,
     // for example, here supports '_',
     // which is a valid character in POSIX-compliant system, like HDFS.
-    return (c == '.' || c == '-' ||
-        Character.isLowerCase(c) || Character.isDigit(c)) ||
-        (c == '_' && !isStrictS3);
+    int code = c;
+    if (code >= 47 && code <= 57) { // 0 - 9
```

Comment:
Why not just use `char` here? Also makes the code more readable:

```suggestion
    if (c >= '0' && c <= '9') {
```

Same applies to two other conditions below.

(Also, `0` is decimal 48 in the [ASCII table](https://www.asciitable.com/), not 47.)

Reply:
Well spotted on the ACSII code error. Using the actual characters is better, I agree. I wasn't aware you could do it like this.

Reply:
Thank you for the suggestion, I've changed it to character comparison now